pyccel/codegen/printing/ccode.py

```diff
@@ -587,6 +603,67 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = base.shape
+        allow_negative_indexes = (isinstance(base, IndexedVariable) and \
+                base.internal_variable.allows_negative_indexes)
+        base = self._print(expr.base.label)
```

Comment:
It would be better if you gave this variable a different name then you can keep using the `base` variable defined above
(Also shouldn't it be `self._print(base.label)`? )

Reply:
i found an unexpected behaviour here when trying to assign a view

eg:
`a=np.array([[1,2,3],[4,5,6]])
b = a[0]`
so i need to handle this case first

Reply:
What is the unuxepected behaviour?

Reply:
because in that case `base` is a `Variable` object and don't have a `label` attribute i just need to check some cases to _print them correctly

Reply:
Ah. The `if`/`else` block creating the `base` variable is supposed to ensure that `base` is always a `Variable` I think. Maybe you just need to use `base.name`?

Reply:
yes that's what i did